### PR TITLE
MIME4J-296 Use Path instead of File in MboxIterator

### PR DESCRIPTION
**Pull request by randomnetcat**
As a side effect, this also makes `close` a no-op because the current version keeps around an InputStream unnecessarily.

### CODE REVIEW

File: mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java [hunk 1]
```diff
@@ -76,8 +78,12 @@ private MboxIterator(final File mbox,
         this.DECODER = charset.newDecoder();
         this.mboxCharBuffer = CharBuffer.allocate(MAX_MESSAGE_SIZE);
         this.mbox = mbox;
-        this.theFile = new FileInputStream(mbox);
-        this.byteBuffer = theFile.getChannel().map(FileChannel.MapMode.READ_ONLY, 0, theFile.getChannel().size());
+
+        try (FileChannel channel = FileChannel.open(mbox, StandardOpenOption.READ)) {
+            // Safe to keep buffer around because it does not depend on the channel.
+            this.byteBuffer = channel.map(FileChannel.MapMode.READ_ONLY, 0, channel.size());
```
Comment (mbaechler) [inline mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java:84]: I never used that API but when will the resource be disposed then?

**Comment by mbaechler** [comment]
Is there any test related to this code?

File: mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java (hunk 1 above)
Comment (randomnetcat) [reply mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java:84]: The `ByteBuffer` is disposed of when garbage collected. There is no explicit way of releasing it:

From https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/nio/channels/FileChannel.html#map(java.nio.channels.FileChannel.MapMode,long,long) :
> The buffer and the mapping that it represents will remain valid until the buffer itself is garbage-collected.

**Comment by randomnetcat** [comment]
There is the existing `MboxIteratorTest` which tests all the same code paths that construction from a `Path` would test.

File: mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java (hunk 1 above)
Comment (mbaechler) [reply mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java:84]: Thank you for the link and the explanation

**Review by mbaechler — APPROVED** [review]
Thank you for your contribution.

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Can you open a JIRA ticket for this?

https://issues.apache.org/jira/projects/MIME4J/issues/MIME4J-250?filter=allopenissues

Then we usually reword the commits to prefix it with the JIRA: `MIME4J-XXX My awesome commit message`

Thanks by advance.

**Comment by randomnetcat** [comment]
Done.

**Comment by chibenwa** [comment]
I will run tests locally then merge this. Give me 15 minutes...

**Comment by chibenwa** [comment]
Tests pass.

Thank you for this contribution.

**Comment by chibenwa** [comment]
I just pushed a rebased version where I reworded the commit with the ticket prefix.

**Pull request merged**